### PR TITLE
r-modules: fix R packages

### PR DESCRIPTION
**Pull request by PhDyellow**
###### Motivation for this change
Some R packages were not building and installing.  

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PhDyellow** [comment]
https://hydra.nixos.org/jobset/nixpkgs/r-updates shows other packages that need fixing.

File: pkgs/development/r-modules/default.nix [hunk 1]
```diff
@@ -948,6 +953,18 @@ let
...
+    R_cache = old.R_cache.overrideDerivation (attrs: {
+      preConfigure = ''
+        export R_CACHE_ROOTPATH=$TMP
+      '';
+    });
+
+    lpsymphony = old.lpsymphony.overrideDerivation (attrs: {
+      preConfigure = ''
+        patchShebangs configure
+        '';
```
Comment (SuperSandro2000) [inline pkgs/development/r-modules/default.nix:965]:
```suggestion
        patchShebangs configure
      '';
```

File: pkgs/development/r-modules/default.nix [hunk 2]
```diff
@@ -412,6 +416,7 @@ let
     openssl = [ pkgs.pkg-config ];
     pdftools = [ pkgs.pkg-config ];
     sf = [ pkgs.pkg-config pkgs.sqlite.dev pkgs.proj.dev ];
+    terra = [ pkgs.pkg-config pkgs.sqlite.dev pkgs.proj.dev ];
```
Comment (SuperSandro2000) [inline pkgs/development/r-modules/default.nix:419]:
```suggestion
    terra = with pkgs; [ pkg-config sqlite.dev proj.dev ];
```

File: pkgs/development/r-modules/default.nix [hunk 3]
```diff
@@ -338,12 +340,14 @@ let
     seqinr = [ pkgs.zlib.dev ];
     seqminer = [ pkgs.zlib.dev pkgs.bzip2 ];
     sf = [ pkgs.gdal pkgs.proj pkgs.geos ];
+    terra = [ pkgs.gdal pkgs.proj pkgs.geos ];
```
Comment (SuperSandro2000) [inline pkgs/development/r-modules/default.nix:343]:
```suggestion
    terra = with pkgs; [ gdal proj geos ];
```

File: pkgs/development/r-modules/default.nix [hunk 4]
```diff
@@ -275,7 +276,8 @@ let
     jqr = [ pkgs.jq.dev ];
     KFKSDS = [ pkgs.gsl_1 ];
     kza = [ pkgs.fftw.dev ];
-    lwgeom = [ pkgs.gdal pkgs.geos pkgs.proj ];
+    lpsymphony = [ pkgs.pkgconfig pkgs.gfortran pkgs.gettext ];
+    lwgeom = [pkgs.proj.dev pkgs.proj pkgs.pkg-config pkgs.geos pkgs.sqlite pkgs.gdal ];
```
Comment (SuperSandro2000) [inline pkgs/development/r-modules/default.nix:280]:
```suggestion
    lpsymphony = with pkgs; [ pkg-config gfortran gettext ];
    lwgeom = with pkgs; [ proj.dev proj pkg-config geos sqlite gdal ];
```

File: pkgs/development/r-modules/default.nix [hunk 5]
```diff
@@ -338,12 +340,14 @@ let
...
     seqminer = [ pkgs.zlib.dev pkgs.bzip2 ];
     sf = [ pkgs.gdal pkgs.proj pkgs.geos ];
+    terra = [ pkgs.gdal pkgs.proj pkgs.geos ];
     showtext = [ pkgs.zlib pkgs.libpng pkgs.icu pkgs.freetype.dev ];
     simplexreg = [ pkgs.gsl_1 ];
     spate = [ pkgs.fftw.dev ];
     ssanv = [ pkgs.proj ];
     stsm = [ pkgs.gsl_1 ];
     stringi = [ pkgs.icu.dev ];
+    svglite = [ pkgs.libpng.dev ];
```
Comment (collares) [inline pkgs/development/r-modules/default.nix:350]: Seems to need a rebase:

```
error: attribute 'svglite' already defined at ./nixpkgs/pkgs/development/r-modules/default.nix:351:5

       at ./nixpkgs/pkgs/development/r-modules/default.nix:353:5:

          352|     survSNP = [ pkgs.gsl_1 ];
          353|     svglite = [ pkgs.libpng.dev ];
             |     ^
          354|     sysfonts = [ pkgs.zlib pkgs.libpng pkgs.freetype.dev ];
```

Comment (PhDyellow) [reply pkgs/development/r-modules/default.nix]: Latest push should resolve this.

File: pkgs/development/r-modules/default.nix (hunk 4 above)
Comment (PhDyellow) [reply pkgs/development/r-modules/default.nix]: I have fixed the repeated `pkgs.` everywhere it came up in `r-modules/default.nix`

**Comment by PhDyellow** [comment]
If the checks pass, I will squash the commits.

**Comment by collares** [comment]
I confirm that this fixes `exactextractr`, `lpsymphony` and `terra`. However, `nix-shell -p 'rWrapper.override{ packages = [ rPackages.fftwtools rPackages.lwgeom ]; }' --pure` seems to build fine even without these changes. How do I test that the changes to `fftwtools` and `lwgeom` are needed? Also, are there any other packages that changed? It's hard to tell with the interspersed `with pkgs` refactoring.

File: pkgs/development/r-modules/default.nix [hunk 6]
```diff
@@ -237,54 +237,56 @@ let
...
-    igraph = [ pkgs.gmp pkgs.libxml2.dev ];
+    igraph = with pkgs; [ gmp libxml2.dev ];
     JavaGD = [ pkgs.jdk ];
     jpeg = [ pkgs.libjpeg.dev ];
     jqr = [ pkgs.jq.dev ];
     KFKSDS = [ pkgs.gsl_1 ];
     kza = [ pkgs.fftw.dev ];
-    lwgeom = [ pkgs.gdal pkgs.geos pkgs.proj ];
+    lpsymphony = with pkgs; [ pkg-config gfortran gettext ];
+    lwgeom = with pkgs; [pkgs.proj.dev proj pkg-config geos sqlite gdal ];
```
Comment (collares) [inline pkgs/development/r-modules/default.nix:280]: There's missing whitespace (after the `[`) here.

**Comment by PhDyellow** [comment]
For `fftwtools`, I made the change because of the hydra build log which showed fftwtools breaking without pkg-config: https://hydra.nixos.org/build/146252915/log/tail

lwgeom seems to be a mixup at my end. I'll remove it from this PR and see if it builds for me. 

Other packages I have attempted to fix are
- `r_import` - there was a naming collision with Nix, 
- `R_cache` - there is another PR to fix R_cache,  #129746. Here I just patch R_cache, #129746 is more general and creates a new override category so any future packages that expect a home directory can be easily fixed.

**Review by collares — APPROVED** [review]
I think the approach in the PR you mentioned is also fine, but personally I'd rather wait until there are at least a few packages with the same requirements before introducing a more general solution. There is no special override category for `patchShebangs configure`, for example, and that certainly happens a whole lot more.

Therefore, I'm happy with the current changes once the `lwgeom` change is double-checked. Please post a comment when this PR is ready so we don't miss it :)

**Comment by PhDyellow** [comment]
- [X] r_import
- [X] R_cache
- [X] fftwtools -breaks when I build from `nixos-unstable`, I'm not sure why it builds for @collares 
```
building '/nix/store/m3aqz28wgvwaqmn22gy67j7k362qm11h-r-fftwtools-0.9-11.drv'...
unpacking sources
unpacking source archive /nix/store/2gljkqnpkg7ixbbh45km898w9jddpr8s-fftwtools_0.9-11.tar.gz
source root is fftwtools
setting SOURCE_DATE_EPOCH to timestamp 1614607803 of file fftwtools/MD5
patching sources
configuring
building
running tests
installing
* installing *source* package 'fftwtools' ...
** package 'fftwtools' successfully unpacked and MD5 sums checked
** using staged installation
checking for gcc... /nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0/bin/cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0/bin/cc accepts -g... yes
checking for /nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0/bin/cc option to accept ISO C89... none needed
checking how to run the C preprocessor... /nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0/bin/cc -E
checking for gcc... (cached) /nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0/bin/cc
checking whether we are using the GNU C compiler... (cached) yes
checking whether /nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0/bin/cc accepts -g... (cached) yes
checking for /nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0/bin/cc option to accept ISO C89... (cached) none needed
checking for pkg-config... no
configure: error: pkg-config not found.
ERROR: configuration failed for package 'fftwtools'
* removing '/nix/store/k356day6v0fw07gfgd3cfw1jgx5rzn09-r-fftwtools-0.9-11/library/fftwtools'
```
- [ ] lwgeom - lwgeom was working on nixos-unstable, so I have reverted the changes to lwgeom. It still appears in the changelog because of the `with pkgs; [` reformatting.

**Comment by PhDyellow** [comment]
`lwgeom` has been reverted, and I have confirmed that the following packages are broken on `nixos-unstable` but fixed in this PR:
- `fftwtools`
- `r_import`
- `R_cache`

@collares has confirmed that the other packages affected by this PR were broken and are now fixed.

**Comment by PhDyellow** [comment]
Ready to merge!

**Review by collares — APPROVED** [review]
I am happy with this PR, but unfortunately I don't have commit privileges. Hopefully @peti, @SuperSandro2000 or @7c6f434c can help us.

**Comment by PhDyellow** [comment]
@jbedo and @bcdarwin are codeowners now, taking over from @peti. Hopefully they can help merge this.

**Comment by bcdarwin** [comment]
I don't have write access to the repository so I can't help, sorry.  (Presumably I should nominate myself for either write access or being removed as CODEOWNER, though.)

**Comment by collares** [comment]
@7c6f434c Thank you!

**Pull request merged**